subsys/bluetooth/mesh/adv_ext.c

```diff
@@ -71,33 +71,26 @@ static bool schedule_send(struct bt_mesh_ext_adv *ext_adv);
 
 static struct bt_mesh_ext_adv advs[] = {
 	[0] = {
-		.tags = (
-#if !defined(CONFIG_BT_MESH_ADV_EXT_FRIEND_SEPARATE)
-			BT_MESH_ADV_TAG_BIT_FRIEND |
-#endif
-#if !defined(CONFIG_BT_MESH_ADV_EXT_GATT_SEPARATE)
-			BT_MESH_ADV_TAG_BIT_PROXY |
-#endif /* !CONFIG_BT_MESH_ADV_EXT_GATT_SEPARATE */
-#if defined(CONFIG_BT_MESH_ADV_EXT_RELAY_USING_MAIN_ADV_SET)
-			BT_MESH_ADV_TAG_BIT_RELAY |
-#endif /* CONFIG_BT_MESH_ADV_EXT_RELAY_USING_MAIN_ADV_SET */
-#if defined(CONFIG_BT_MESH_PB_ADV)
-			BT_MESH_ADV_TAG_BIT_PROV |
-#endif /* CONFIG_BT_MESH_PB_ADV */
-			BT_MESH_ADV_TAG_BIT_LOCAL
+		.tags = ( BT_MESH_ADV_TAG_BIT_LOCAL |
+			  IS_ENABLED(CONFIG_BT_MESH_ADV_EXT_FRIEND_SEPARATE) ?
+				0 : BT_MESH_ADV_TAG_BIT_FRIEND |
```

Comment:
Are you sure parentheses are not needed here? Ternary operator has lower precedence than bitwise OR: https://en.cppreference.com/w/c/language/operator_precedence

Reply:
I'm not sure, maybe look preprocessing .i file, i still don't know how to view the .i file compiled by zephyr. But add parentheses always right.